client/client-lib/src/lib.rs

```diff
@@ -1111,8 +1106,8 @@ impl Client<GatewayClientConfig> {
 
         // Inputs
         let mut builder = TransactionBuilder::default();
-        let coins = self.mint_client().select_coins(offer.amount).await?;
-        builder.input_coins(coins)?;
+        let (mut keys, input) = self.mint_client().select_input(offer.amount).await?;
+        builder.input(&mut keys, input);
```

Comment:
nit: We might want a generic spendable input type so we don't need to destructure the tuple here.

```rust
struct SpendableInput<I> {
    keys: Vec<KeyPair>,
    input: I,
}
```